dds_web/scheduled_tasks.py

```diff
@@ -1,24 +1,74 @@
 import flask_apscheduler
 import flask
 
-# Apscheduler
+## Apscheduler
 scheduler = flask_apscheduler.APScheduler()
 
 
+# @scheduler.task("cron", id="available_to_expired", minute=5, hour=2, misfire_grace_time=1)
 @scheduler.task("interval", id="available_to_expired", seconds=10, misfire_grace_time=1)
-def change_status_to_expired():
-    # print("Task: Change project status from Available to expired.", flush=True)
-    from dds_web.database import models
+def set_available_to_expired():
+    scheduler.app.logger.debug("Task: Checking for Expiring projects.")
+    import sqlalchemy
 
-    # from dds_web.utils import current_time, page_query
+    from dds_web import db
+    from dds_web.database import models
+    from dds_web.errors import DatabaseError
+    from dds_web.api.project import ProjectStatus
+    from dds_web.utils import current_time, page_query
 
     with scheduler.app.app_context():
-        # scheduler.app.logger.debug("This means that the app context works!")
-        test = models.Project.query.all()
-        # print(test, flush=True)
+        expire = ProjectStatus()
+
+        errors = {}
+
+        for unit in models.Unit.query.all():
+            errors[unit.name] = {}
+
+            days_in_expired = unit.days_in_expired
+
+            for project in page_query(
+                models.Project.query.filter(
+                    sqlalchemy.and_(
+                        models.Project.is_active == 1, models.Project.unit_id == unit.id
+                    )
+                )
```

Comment:
Same thing here with the blocking
```suggestion
                db.session.query(models.Project).filter(
                    sqlalchemy.and(
                        models.Project.is_active == 1, models.Project.unit_id == unit.id
                    )
                ).with_for_update()
```